Branch planner NPE because it looked for `spec.writeOutputsToSecret.name`, while `spec.writeOutputsToSecret` is an optional field

Branch planner NPE because it looked for spec.writeOutputsToSecret.name, while spec.writeOutputsToSecret is an optional field.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1fee5dc]
goroutine 27 [running]:
github.com/weaveworks/tf-controller/internal/server/polling.(*Server).reconcileTerraform(0xc0002a6680, {0x257b528, 0xc00004f180}, 0xc0002ac500, 0xc0007b05a0, {0xc000a0d460, 0x6}, {0x3000ce8, 0x1}, 0x6fc23ac00)
    /workspace/internal/server/polling/terraform.go:95 +0x5bc
github.com/weaveworks/tf-controller/internal/server/polling.(*Server).reconcile(0xc0002a6680, {0x257b528, 0xc00004f180}, 0xc0002ac500, 0xc0007b05a0, {0xc000468080, 0x1, 0x1}, {0x257f528, 0xc0005962c0})
    /workspace/internal/server/polling/server.go:184 +0x6cd
github.com/weaveworks/tf-controller/internal/server/polling.(*Server).poll(0xc0002a6680, {0x257b528, 0xc00004f180}, {{0xc000786840, 0xb}, {0xc000595488, 0x13}}, 0xc00047c640)
    /workspace/internal/server/polling/server.go:167 +0x1573
github.com/weaveworks/tf-controller/internal/server/polling.(*Server).Start(0xc0002a6680, {0x257b528, 0xc00004f180})
    /workspace/internal/server/polling/server.go:115 +0xfb5
main.startPollingServer({0x257b528, 0xc00004f180}, {{0x257cd20, 0xc00045de00}, 0x0}, {0x2581120, 0xc0000d27e0}, 0xc000790240)
    /workspace/cmd/branch-planner/polling.go:25 +0x3e8
main.main.func1({{0x257cd20, 0xc00045de00}, 0x0})
    /workspace/cmd/branch-planner/main.go:50 +0xe8
created by main.main
    /workspace/cmd/branch-planner/main.go:47 +0x657
```

--- a/internal/server/polling/terraform.go
+++ b/internal/server/polling/terraform.go
@@ -91,9 +91,8 @@ func (s *Server) reconcileTerraform(ctx context.Context, originalTF *infrav1.Ter
 			Namespace: originalTF.Namespace,
 		},
 	}
-	branchLabels := s.createLabels(originalTF.Labels, originalTF.Name, branch, prID)
-	branchSecretName := s.createObjectName(originalTF.Spec.WriteOutputsToSecret.Name, branch, prID)
 
+	branchLabels := s.createLabels(originalTF.Labels, originalTF.Name, branch, prID)
 	op, err := controllerutil.CreateOrUpdate(ctx, s.clusterClient, tf, func() error {
 		spec := originalTF.Spec.DeepCopy()
 
@@ -101,7 +100,10 @@ func (s *Server) reconcileTerraform(ctx context.Context, originalTF *infrav1.Ter
 		spec.SourceRef.Namespace = source.Namespace
 		spec.PlanOnly = true
 		spec.StoreReadablePlan = "human"
-		spec.WriteOutputsToSecret.Name = branchSecretName
+		// relocate the output secret, so it's not shared between branches
+		if spec.WriteOutputsToSecret != nil && originalTF.Spec.WriteOutputsToSecret != nil {
+			spec.WriteOutputsToSecret.Name = s.createObjectName(originalTF.Spec.WriteOutputsToSecret.Name, branch, prID)
+		}
 		spec.ApprovePlan = ""
 		spec.Force = false
 

